docs/developer/index.rst

```diff
@@ -698,6 +698,51 @@ Notes
 
 Notes on the design and changes to Sherpa.
 
+.. _handling_nd:
+
+N-dimensional data and models
+-----------------------------
+
+Models and data objects are
+designed to work with flattened arrays, so a 1D dataset has ``x`` and
+``y`` for the independent and dependent axes, and a 2D dataset will
+have ``x0``, ``x1``, and ``y`` values, with each value stored as a 1D
+`ndarray`. This makes it easy to deal with filters and sparse or
+irregularly-placed grids.
+
+::
+
+  >>> from sherpa.data import Data1D, Data1DInt, Data2D
+
+As examples, we have a one-dimension dataset with data values
+(dependent axis, y) of 2.3, 13.2, and -4.3 corresponding to the
+independent axis (x) values of 1, 2, and 5::
+
+  >>> d1 = Data1D("ex1", [1, 2, 5], [2.3, 13.2, -4.3])
+
+An "integrated" one-dimensional dataset for the independent axis
+bins 23-44, 45-50, 50-53, and 55-57, with data values of
+12, 14, 2, and 22:
+
+  >>> d2 = Data1DInt("ex2", [23, 45, 50, 55], [44, 50, 53, 57], [12, 14, 2, 22])
+
+An irregularly-gridded 2D dataset, with points at (-200, -200),
+(-200, 0), (0, 0), (200, -100), and (200, 150) can be created
+with:
+
+  >>> d3 = Data2D("ex3", [-200, -200, 0, 200, 200], [-200, 0, 0, -100, 150],
+  ... [12, 15, 23, 45, -2])
+
+A regularly-gridded 2D dataset can be created, but note that the
+arguments must be flattened:
+
+  >>> x1, x0 = np.mgrid[20:30:2, 5:20:2]
+  >>> shp = x0.shape
+  >>> x0 = x0.flatten()
+  >>> x1 = x1.flatten()
+  >>> y = np.sqrt((x0 - 10)**2 + (x1 - 31)**2)
+  >>> d4 = Data2D("ex4", x0, x1, y, shape=shp)
```

Comment:
If we have 2D data for x, we typically also have 2D data for y.  Here, we just make that data for the example, but in a typical use case, we would get the x  and y in some other way (e.g. read from a file) and they wuld both be 2d to begin with.
```suggestion
  >>> y = np.sqrt((x0 - 10)**2 + (x1 - 31)**2)
  >>> x0 = x0.flatten()
  >>> x1 = x1.flatten()
  >>> y = y.flatten()
  >>> d4 = Data2D("ex4", x0, x1, y, shape=shp)
```

Reply:
I don't really see this as an improvement but as I wrote it I may biased...